Password reset page

supsshinde opened the pull request:
## Password Reset Page

## Description
Implemented an OTP-based Password Reset page using Angular. This page provides a simple and user-friendly interface that allows users to securely reset their password by verifying their identity with a One-Time Password (OTP).
The page includes the following features:
 - Accepts a registered username or email to initiate the password reset process.
 - Sends an OTP to the registered email for verification.
 - Accepts and validates the OTP entered by the user before allowing password reset.
 - Accepts a new password and confirmation of the new password after successful OTP verification.
Performs form validations, including:
 - Required field checks
 - Password and confirm password matching
 - OTP field validation
 - Displays clear error messages for invalid input (e.g., incorrect OTP, mismatched passwords) and success messages upon successful password update.

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Project Configuration
- [ ] Repository Configuration
- [ ] Documentation update - Code guide/ docs (_like, javadoc or api_).
- [ ] Other (please describe):

## Screenshots / GIFs / Media (if applicable)
- Forgot Password Email:
<img width="926" height="860" alt="account recovery" src="https://github.com/user-attachments/assets/4325bc73-ca68-438c-97ac-eb9c0d2b5200" />

## How Has This Been Tested?/ Test Plan
 - It has been manually tested by running the Angular frontend on the development server using ng serve.
 - Manual tests were performed for the complete password reset flow, including:
 - Entering username to request password reset
 - Verifying the OTP received
 - Submitting the new password
 - Additional integration testing was done by connecting the frontend with the backend project to confirm that the password is          successfully updated in the system.

## Checklist:
- [x] My code follows the project’s [coding guidelines](./../../docs/guidelines/coding-guides/). 
- [x] I have performed a self-review of my own code.
- [x] I have commented on my code where necessary.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [ ] I have added tests that verify the functionality of the code (if applicable).
- [x] All new and existing tests pass.

## How to use/ enable
Open the browser and go to:
 http://localhost:4200/
Step 1 – Identify
 -  Enter your registered email or username.
 - Click “Send OTP”.
 - Success message confirms OTP sent (masked email may display).
Step 2 – Verify OTP
 - Check email for 6-digit OTP.
 - Enter OTP.
Options:
 - Change email → go back to Step 1
 - Resend OTP → request a new OTP
Click “Verify OTP” → navigates to Step 3 on success.
 - Step 3 – Reset Password
 - Enter New Password and Confirm Password.
Validations:
 - Minimum 6 characters
 - Passwords must match
 - Click “Reset Password” → success message appears.

## Additional Notes
- In service.ts file inside change url as per backend API. 
-  Follow steps from [Frontend Setup](./../tree/dev-main/Frontend), if setup is incomplete or need to do fresh setup to run Frontend project.

## Closing issues
closes #172

Commit:
149-us-user-login--dashboard1

created login page

supsshinde committed:
Updated Password Reset Page

Update Password Page
- Revert all unnecessary files and Backend files.
- Code changed
- Created more attractive form
- Remove unnecessary comments and blank lines
- Unnecessary component deleted like Dashboard, Login, Auth etc.

ajstyle2410 commented:
1. update the pull request questions 

- Please check if the PR fulfills these requirements
- What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
- What is the current behavior?** (You can also link to an open issue here)
- What is the new behavior (if this is a feature change)?
- Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
- Test Plan

2. enter the alternate data to show the confirmed password and that the password are working but replace the same password and confirm that the password does not load the angular component.ts file inside  
3. remove the unwanted lines like as comment code

ajstyle2410 commented on Frontend/Arc-i-Tech/src/app/reset-pass/reset-pass.component.html:
```diff
@@ -0,0 +1,47 @@
+                required
+            />
+
+            <!-- Error message for validation -->
```
remove the unwanted comment lines

supsshinde replied:
Unwanted comments and lines removed

ajstyle2410 commented:
@supsshinde    
update the pull request template

supsshinde commented:
@ajstyle2410 review the PR, It's updated.

supsshinde commented:
@ajstyle2410 PR updated review it

ajayg2808 commented:
@supsshinde,

Update description.

supsshinde commented:
> @supsshinde,
> 
> Update description.

Description updated.

ajayg2808 commented on .metadata/.plugins/org.eclipse.core.runtime/.settings/org.eclipse.jdt.launching.prefs:
Why java PATH is modified?
Do not commit any files in **.metadata/**

supsshinde replied:
@ajayg2808 All unnecessary files reverted

ajayg2808 commented on .metadata/.plugins/org.eclipse.core.resources/3.snap:
Do not commit any file modifications from **./metadata**

supsshinde replied:
> Do not commit any file modifications from **./metadata**

@ajayg2808  ./metadata file reverted

supsshinde commented:
@ajayg2808 PR description updated.

ajayg2808 reviewed (changes requested):
Revert below changes:
- [x] **.metadata/**
- [x] **Arc-i-Tech/** (As you are working on frontend, any back end changes (even related workspace) are not expected.

ajayg2808 commented:
@supsshinde,
Waiting for your updates.

supsshinde commented:
> Revert below changes:
> 
> * [ ]  **.metadata/**
> * [ ]  **Arc-i-Tech/** (As you are working on frontend, any back end changes (even related workspace) are not expected.



> Member

I'm trying to revert but some errors occurred like fatal error.

supsshinde commented:
@ajayg2808 sir in this PR if any changes required then revert me back

ajayg2808 commented:
@supsshinde 
Do below updates
- [x] Assign PR
- [x] Description explain how to use this functionality, instead it should be describing what you have implemented in this PR (what feature)
- [x] Test plan is incorrect
- [x] How to use/ enable is incorrect

supsshinde commented:
@ajayg2808 I didn't get meaning of Assign PR ?

ajayg2808 commented:
> @ajayg2808 I didn't get meaning of Assign PR ?

How we assign an issue the same way you can assign a pull request to its responsible.

supsshinde commented:
@ajayg2808 PR updated.

ajayg2808 reviewed (changes requested):
Do below updates
- [ ] Remove additional commit from @shubham-8551 
- [ ] Reset password URL should be `*/reset-password`
- [x] Create form like: [Contact Us](https://arc-i-tech.github.io/contact.html)
- [ ] Add a proper validations
- [ ] Use color highlighting for valid or invalid field values

supsshinde committed:
password reset page

Updated

supsshinde committed:
Revert "password reset page"

This reverts commit 38ef467e20f9466eea4c1176db164ea1a6d631d7.

priyapal2000 committed:
Update Password Reset Page

Updated the html pages
1. send reset email url page
2. create the reset password page
3. write on the component.ts file inside write the logic and send to service.ts file inside

ajstyle2410 commented:
- [x] 1. remove the unwanted committed files
- [x] 2. remove the unnecessary console.log messages
- [x] 3. change the description on pull request inside 
- [x] 4. update the  Test plan and How to Use

priyapal2000 committed:
Delete Frontend/Arc-i-Tech/src/app/app.component.spec.ts

delete the not necessary file.

priyapal2000 committed:
Delete Frontend/Arc-i-Tech/src/app/password.service.spec.ts

Not necessary this spec.ts file

priyapal2000 committed:
Delete Frontend/Arc-i-Tech/src/app/reset-pass/reset-pass.component.spec.ts

priyapal2000 committed:
Delete Frontend/Arc-i-Tech/src/Users/services/auth-service.service.spec.ts

priyapal2000 committed:
Merge branch '149-us-user-login-dashboard' into 172-t-password-reset-page

priyapal2000 committed:
updated the all files.

update the all frontend files and merge the reset-password-api branch .

priyapal2000 commented:
@ajayg2808  please review this task.
